src/User.php

```diff
@@ -23,6 +23,37 @@ class User
      *
      * @return string
      */
+
+    /**
+    * Unique identifier
+    * @var string
+    */
+    protected $username;
+
+    /**
+     * User email
+     *
+     * @var string
+     */
+    protected $email;
+
+    public function __construct(array $args)
+    {
+        if (isset($args['name'])) {
```

Comment:
@j-feng13 it kind of looks like this can be done in a map:
```php
foreach (['name', 'photoUrl', 'username', 'email'] as $attr) {
  if (isset($args[$attr]) {
    $this->$attr = $args[$attr];
  }
}
```
up to you though, both ways work